core/src/main/java/com/linecorp/armeria/server/annotation/decorator/LoggingDecoratorFactoryFunction.java

```diff
@@ -38,15 +39,23 @@ public final class LoggingDecoratorFactoryFunction implements DecoratorFactoryFu
         final float failureSamplingRate =
                 parameter.failureSamplingRate() >= 0.0f ? parameter.failureSamplingRate()
                                                         : parameter.samplingRate();
-        return LoggingService.builder()
-                             .logWriter(LogWriter.builder()
-                                                 .requestLogLevel(parameter.requestLogLevel())
-                                                 .successfulResponseLogLevel(
-                                                         parameter.successfulResponseLogLevel())
-                                                 .failureResponseLogLevel(parameter.failureResponseLogLevel())
-                                                 .build())
-                             .successSamplingRate(successSamplingRate)
-                             .failureSamplingRate(failureSamplingRate)
-                             .newDecorator();
+        final LogWriter logWriter = LogWriter.builder()
+                                             .requestLogLevel(parameter.requestLogLevel())
+                                             .successfulResponseLogLevel(parameter.successfulResponseLogLevel())
+                                             .failureResponseLogLevel(parameter.failureResponseLogLevel())
+                                             .build();
+        final LoggingServiceBuilder builder
+                = LoggingService.builder()
+                                .logWriter(logWriter)
+                                .successSamplingRate(successSamplingRate)
+                                .failureSamplingRate(failureSamplingRate);
+
+        if (parameter.slowRequestSamplingPercentile() >= 0.0f) {
```

Comment:
Should users simply use the hard limit `slowRequestSamplingUpperBoundMillis` without percentile?

Reply:
Sure, I think it makes sense 👍